### PR TITLE
refactor: use graasp sdk

### LINKED ISSUE #9: Use graasp sdk

### PR DESCRIPTION
**Pull request by pyphilia**
close #9

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 16.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: 16.x
+      - name: Get npm cache directory
+        id: npm-cache
+        run: |
+          echo "::set-output name=dir::$(npm config get cache)"
```
Comment (mochiruntime) [inline .github/workflows/test.yml]: Does yarn use the npm cache? :thinking:

File: src/task-manager.ts [hunk 2]
```diff
@@ -30,15 +23,14 @@ export class TaskManager implements ItemFlagTaskManager {
...
-      this.itemService,
-      this.itemMembershipService,
-      this.itemFlagService,
-    );
+  createCreateTaskSequence(
+    member: Member,
+    data: Partial<ItemFlag>,
+    itemId: string,
+  ): Task<Actor, unknown>[] {
+    const tasks = this.itemTaskManager.createGetTaskSequence(member, itemId);
```
Comment (mochiruntime) [inline src/task-manager.ts:31]: This is just to check the access right? That's why `t3.getInput` is not needed

File: src/tasks/create-item-flag-task.ts [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
-import { DatabaseTransactionHandler } from 'graasp';
-// other services
-import { Member, ItemService, ItemMembershipService } from 'graasp';
-// local
-import { ItemNotFound, MemberCannotReadItem, FlagNotFound } from '../util/graasp-item-flags-error';
-import { ItemFlagService } from '../db-service';
-import { BaseItemFlagTask } from './base-item-flag-task';
+
+import { DatabaseTransactionHandler, TaskStatus } from '@graasp/sdk';
+import { Member } from '@graasp/sdk';
```
Comment (mochiruntime) [inline src/tasks/create-item-flag-task.ts]: Combine imports?
```suggestion
import { DatabaseTransactionHandler, Member, TaskStatus } from '@graasp/sdk';
```

File: src/util/graasp-item-flags-error.ts [hunk 4]
```diff
@@ -1,35 +1,9 @@
...
-  }
-}
-
-export class MemberCannotReadItem extends GraaspItemFlagsError {
-  constructor(data?: unknown) {
-    super({ code: 'GIFERR002', statusCode: 403, message: 'Member cannot read item' }, data);
-  }
-}
-export class FlagNotFound extends GraaspItemFlagsError {
+export const GraaspError = ErrorFactory(PLUGIN_NAME);
```
Comment (mochiruntime) [inline src/util/graasp-item-flags-error.ts]:
```suggestion
export const GraaspItemFlagsError = ErrorFactory(PLUGIN_NAME);
```

File: src/util/graasp-item-flags-error.ts [hunk 5]
```diff
@@ -1,35 +1,9 @@
...
+export class FlagNotFound extends GraaspError {
```
Comment (mochiruntime) [inline src/util/graasp-item-flags-error.ts]:
```suggestion
export class FlagNotFound extends GraaspItemFlagsError {
```

**Review by mochiruntime — CHANGES_REQUESTED** [review]
Awesome :cat2: I've left a few questions and some style nitpicks

File: src/task-manager.ts (hunk 2 above)
Comment (pyphilia) [reply src/task-manager.ts:31]: It's just to check the access indeed, and t3 doesn't need any data from the access!

**Review by mochiruntime — APPROVED** [review]

**Pull request merged**